public/components/datasources/components/new/configure_datasource.tsx

```diff
@@ -73,13 +94,37 @@ export function Configure(props: ConfigureDatasourceProps) {
             setDetailsForRequest={setDetails}
             currentArn={arn}
             setArnForRequest={setArn}
-            currentStore={store}
-            setStoreForRequest={setStore}
+            currentStore={storeURI}
+            setStoreForRequest={setStoreURI}
             roles={roles}
             selectedQueryPermissionRoles={selectedQueryPermissionRoles}
             setSelectedQueryPermissionRoles={setSelectedQueryPermissionRoles}
           />
         );
+      case 'PROMETHEUS':
+        return (
+          <ConfigurePrometheusDatasource
+            currentName={name}
+            currentDetails={details}
+            setNameForRequest={setName}
+            setDetailsForRequest={setDetails}
+            currentStore={storeURI}
+            setStoreForRequest={setStoreURI}
+            roles={roles}
+            currentUsername={username}
+            setUsernameForRequest={setUsername}
+            currentPassword={password}
+            setPasswordForRequest={setPassword}
+            selectedQueryPermissionRoles={selectedQueryPermissionRoles}
+            setSelectedQueryPermissionRoles={setSelectedQueryPermissionRoles}
+            currentAccessKey={accessKey}
+            currentSecretKey={secretKey}
+            setAccessKeyForRequest={setAccessKey}
+            setSecretKeyForRequest={setSecretKey}
+            currentRegion={region}
+            setRegionForRequest={setRegion}
```

Comment:
Question: Is there a way to reduce the number of props here?

Reply:
This is side effect of this issue: https://github.com/opensearch-project/dashboards-observability/pull/1054#discussion_r1340408705, if there is a cleaner way to do it I am all for it, but I couldn't find one with quick search

Reply:
This wouldn't be an issue if we render the form inline, but rendering the form as a child is causing issues

Reply:
Maybe mis-understood your question here - I don't think there is a way to do so, since I need to know the values in the parent in order to send the request and have a way of updating them in the child. Thus I need to pass in value and setter for each of the possible configuration properties. Consolidating it into an object might be cleaner, but I think this  way makes it clear what is going in and as what. Leaning towards keeping it like this for now